### PR TITLE
175 create a general button component

### LINKED ISSUE #175: Create a general Button Component
- option / property to open links (`:to`)
- property to support lazy mode (async click events) with a spinner animation indicating something is happening and disable it while loading
- option to make the button a back link

### PR DESCRIPTION
**Pull request by schnickalodeon**
closes #175

### CODE REVIEW

File: packages/app/src/views/NotFound.vue [hunk 1]
```diff
@@ -1,17 +1,22 @@
 <template>
   <div class="m-auto">
-    <router-link
+    <!-- <router-link
       class="cursor-pointer bg-green-600 rounded-full text-center p-2 text-white"
       :to="{ name: 'home' }"
       >Back to home</router-link
```
Comment (JanEberlein) [inline packages/app/src/views/NotFound.vue:6]:
```suggestion
```
If this isn't needed any more, please delete it

File: packages/app/src/views/NotFound.vue [hunk 2]
```diff
@@ -1,17 +1,22 @@
...
-    >
+    > -->
```
Comment (JanEberlein) [inline packages/app/src/views/NotFound.vue:7]:
```suggestion
```
github doesn't allow this suggestion to be in one piece 😞

**Review by JanEberlein — CHANGES_REQUESTED** [review]

**Comment by schnickalodeon** [comment]
I added the functionality to show an image next to the text in the button (nessesary for #163)

**Review by JanEberlein — APPROVED** [review]

File: packages/app/src/components/uiBlocks/Button.vue [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  props: {
+    text: {
+      type: String,
+      required: true,
+    },
+    to: {
+      type: [Object, String],
+      default: null,
+    },
+    back: {
```
Comment (tcoro) [inline packages/app/src/components/uiBlocks/Button.vue:27]: I would rename this to clarify what this does, like
```suggestion
    revert: {
```

**Review by tcoro — CHANGES_REQUESTED** [review]
good work, would change that one variable name for clarification

File: packages/app/src/components/uiBlocks/Button.vue (hunk 3 above)
Comment (schnickalodeon) [reply packages/app/src/components/uiBlocks/Button.vue]: @tcoro is changed :+1:

**Review by tcoro — APPROVED** [review]

**Pull request merged**